fsm-library/src/MachineDefinition.js

```diff
@@ -43,15 +43,20 @@ export default class MachineDefinition {
     const checkFrom = transition => {
       return transition.from === from;
     };
+
     const checkEvent = transition => {
-      // if event is not specified then event does not metter for search
+      // if event is not specified then event does not matter for search
       if (!event) {
         return true;
       }
       return transition.event === event;
     };
-    const checkGuards = transition => {
-      const { guards, from, to, event } = transition;
+
+
+    const checkGuards = ({transition, guardKey = 'guards'}) => {
```

Comment:
I don't like the idea of such kind of optimizations. When checking "automatic" transitions I would like to see it just as additional condition like ```.. && checkAutomatic(). Don't spread the logic about automatic transitions through whole method "findAvailableTransitions". Don't optimize the code that is far away from being a production one it is to be changed anyway. I would implement it differently:

```javascript
// inside the "findAvailableTransitions" method
checkAutomatic = (transition) => {
  if (isAutomatic) {
    // we don't need to check "automatic" transitions at all 
     return true;
  }
  // check if transition automatic conditions are fullfilled
  // return true or false
};

let foundTransitions = transitions.filter(transition => {
  return checkFrom(transition) && 
    checkEvent(transition) && 
    checkGuards(transition) &&
    checkAutomatic(transition);		          
});
```